plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,461 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: ndo_ipsla_monitoring_policy
+short_description: Manage IPSLA Monitoring Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage IPSLA Monitoring Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a tenant template.
+    type: str
+    required: true
+  ipsla_monitoring_policy:
+    description:
+    - The name of the IPSLA Monitoring Policy.
+    type: str
+    aliases: [ name ]
+  description:
+    description:
+    - The description of the IPSLA Monitoring Policy.
+    type: str
+  ipsla_monitoring_policy_uuid:
+    description:
+    - The uuid of the IPSLA Monitoring Policy.
+    - This parameter is required when the O(ipsla_monitoring_policy) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  sla_type:
+    description:
+    - The type of SLA.
+    type: str
+    choices: [ http, tcp, icmp, l2ping ]
+  destination_port:
+    description:
+    - The port used for SLA.
```

Comment:
```suggestion
    - The port used for SLA.
    - Only used when O(sla_type=tcp) otherwise it will be ignored.
```
Something like that.